src/main/java/me/jellysquid/mods/sodium/client/render/chunk/DefaultChunkRenderer.java

```diff
@@ -208,25 +246,48 @@ private static float getCameraTranslation(int chunkBlockPos, int cameraBlockPos,
 
     private GlTessellation prepareTessellation(CommandList commandList, RenderRegion region) {
         var resources = region.getResources();
-        var tessellation = resources.getTessellation();
 
+        GlTessellation tessellation = resources.getTessellation();
         if (tessellation == null) {
-            resources.updateTessellation(commandList, tessellation = this.createRegionTessellation(commandList, resources));
+            tessellation = this.createRegionTessellation(commandList, resources);
+            resources.updateTessellation(commandList, tessellation);
+        }
+
+        return tessellation;
+    }
+
+    private GlTessellation prepareIndexedTessellation(CommandList commandList, RenderRegion region) {
+        var resources = region.getResources();
+
+        GlTessellation tessellation = resources.getIndexedTessellation();
+        if (tessellation == null) {
+            tessellation = this.createIndexedRegionTessellation(commandList, resources);
+            resources.updateIndexedTessellation(commandList, tessellation);
         }
 
         return tessellation;
     }
 
     private GlTessellation createRegionTessellation(CommandList commandList, RenderRegion.DeviceResources resources) {
         return commandList.createTessellation(GlPrimitiveType.TRIANGLES, new TessellationBinding[] {
-                TessellationBinding.forVertexBuffer(resources.getVertexBuffer(), new GlVertexAttributeBinding[] {
+                TessellationBinding.forVertexBuffer(resources.getGeometryBuffer(), new GlVertexAttributeBinding[] {
                         new GlVertexAttributeBinding(ChunkShaderBindingPoints.ATTRIBUTE_PACKED_DATA,
                                 this.vertexFormat.getAttribute(ChunkMeshAttribute.VERTEX_DATA))
                 }),
                 TessellationBinding.forElementBuffer(this.sharedIndexBuffer.getBufferObject())
         });
     }
 
+    private GlTessellation createIndexedRegionTessellation(CommandList commandList, RenderRegion.DeviceResources resources) {
```

Comment:
Can this be implemented as a `useSharedIndex` argument to `createRegionTessellation` instead?